pkg/signup/service/signup_service_test.go

```diff
@@ -2104,6 +2104,73 @@ func (s *TestSignupServiceSuite) TestGetSignupUpdatesUserSignupAnnotations() {
 		require.Equal(s.T(), "888888", modified.Annotations[toolchainv1alpha1.SSOUserIDAnnotationKey])
 		require.Equal(s.T(), "1234567890", modified.Annotations[toolchainv1alpha1.SSOAccountIDAnnotationKey])
 	})
+
+	s.Run("PreferredUsername property updated when set in context", func() {
+		c, _ := gin.CreateTestContext(httptest.NewRecorder())
+		c.Set(context.UsernameKey, "cocochanel")
+
+		_, err := s.Application.SignupService().GetSignup(c, userSignup.Name, userSignup.Spec.Username)
+		require.NoError(s.T(), err)
+
+		modified, err := s.FakeUserSignupClient.Get(userSignup.Name)
+		require.NoError(s.T(), err)
+
+		require.Equal(s.T(), "cocochanel", modified.Spec.IdentityClaims.PreferredUsername)
+	})
+
+	s.Run("GivenName property updated when set in context", func() {
+		c, _ := gin.CreateTestContext(httptest.NewRecorder())
+		c.Set(context.GivenNameKey, "Jonathan")
+
+		_, err := s.Application.SignupService().GetSignup(c, userSignup.Name, userSignup.Spec.Username)
+		require.NoError(s.T(), err)
+
+		modified, err := s.FakeUserSignupClient.Get(userSignup.Name)
+		require.NoError(s.T(), err)
+
+		require.Equal(s.T(), "Jonathan", modified.Spec.IdentityClaims.GivenName)
+
+		// Confirm that some other properties were not changed
+		require.Equal(s.T(), "cocochanel", modified.Spec.IdentityClaims.PreferredUsername)
```

Comment:
This test depends on the previous `"PreferredUsername property updated when set in context"` so maybe worth moving it under that test as a sub-test. But I guess it's fine to keep it this way too. Maybe worth adding a comment that it's coming from the previous test.

Reply:
good catch, i'll move it under the other test